idpf_specification.md

```diff
@@ -6433,28 +6433,101 @@ VIRTCHNL2_OP_RDMA			= 529,
+a.	In legacy devices with no PCIE PTM capability this was done by independently sampling device time and system time but doing it in fashion that they can be done as close in time as possible or by taking two system time sample one before taking the device time sample and one immediately after the device time sample and the mean system time value between the two would be approximately the same time when the device time was sampled.
+This capability is exposed as 
+VIRTCHNL2_PTP_CAP_GET_DEVICE_CLK_TIME
+VIRTCHNL2_PTP_CAP_GET_DEVICE_CLK_TIME_MB
+Note: The Mailbox (_MB) version to get Device time over mailbox is generally not in use but is for completeness as the Device time over mailbox cannot be relied as being done exactly in the middle of the two system time samples or due to proxy nature can be far off from the system time sample. The capability returns only one time value from the Device, that of the Device master time.
+b.	With PCIE PTM capability, the system OS time and the Device time can be latched simultaneously by the device. This is done by the Device’s ability to sample the system ART timer at the same time the device master timer is sampled. 
+This capability is exposed as
+VIRTCHNL2_PTP_CAP_GET_CROSS_TIME	
+VIRTCHNL2_PTP_CAP_GET_CROSS_TIME_MB 
+Either capability is as accurate as the other in sampling the system and device time as the Mailbox (_MB) mechanism also relies on the Device control plane to be able to latch both system time and the Device at the same instance. The mailbox method just allows for indirect access to reading the latched values by the device. This capability returns two time values one for the Device and one for the system (CPU complex)
+2.	Ability to do Tx time stamping of packets as they leave the Device (typically in the PHY).
+a.	This capability can be used by either the follower (slave)clock server or by the leader (master) clock server in the common PTP network domain to calculate packet delay in transmission and reception between follower and the leader. 
+VIRTCHNL2_PTP_CAP_TX_TSTAMPS	
+VIRTCHNL2_PTP_CAP_TX_TSTAMPS_MB
+The packet delay calculation on the wire is used to accurately remove the time component that is common in calculation of the skew in clock between the leader and the follower. Tx timestamping should be allowed on all vports that are tied to an external port which means there traffic goes only to a particular external physical port.
+3.	Ability to Adjust Device Master Clock, this capability in a multi-complex system is given to one of the CPU complex that is trusted to make this master clock adjustment.  The Complex that calculates the skew between the leader (master system in the PTP network domain)) and the current Device master clock (slave system in the PTP network domain) runs the PTP protocol and algorithm in SW which requires other capabilities such as TX_TSTAMPS and cross timestamping as well from the Device as described in 1 and 2 above. 
```

Comment:
For the adjustments, there're also defined fields dedicated to the PHY. Could you please describe the flow of adjustments? I don't mean the specific flow, rather the possibilities using provided API.

Reply:
Done as part of the flow.